python/cuml/manifold/umap.pyx

```diff
@@ -590,29 +659,51 @@ class UMAP(Base):
             return ret
 
         cdef UMAPParams * umap_params = \
-            <UMAPParams*> <size_t> self.umap_params
+            <UMAPParams * > < size_t > self.umap_params
         embedding = rmm.to_device(zeros((X_m.shape[0],
                                          umap_params.n_components),
                                         order="C", dtype=np.float32))
         cdef uintptr_t xformed_ptr = embedding.device_ctypes_pointer.value
 
-        cdef cumlHandle *handle_ = \
-            <cumlHandle*> <size_t> self.handle.getHandle()
+        knn_indices_ctype = None
+        knn_dists_ctype = None
```

Comment:
This is a complete duplication of the training. Can we pull this into a shared function to minimize the potential for bugs as these are modified and updated?